Intersect.Client.Core/Entities/Resource.cs

```diff
@@ -250,15 +336,9 @@ public override bool Update()
         return result;
     }
 
-    /// <inheritdoc />
-    public override bool CanBeAttacked => !IsDead;
-
     public override HashSet<Entity>? DetermineRenderOrder(HashSet<Entity>? renderList, IMapInstance? map)
     {
-        if (Descriptor == default ||
-            (IsDead && !Descriptor.Exhausted.RenderBelowEntities) ||
-            (!IsDead && !Descriptor.Initial.RenderBelowEntities)
-        )
+        if (Descriptor == default || CurrentState is not { } graphicState || !graphicState.RenderBelowEntities)
```

Comment:
We aren't using `Descriptor` at all in this if statement, I think it should be removed from the check here